aio_pika/robust_connection.py

```diff
@@ -77,44 +91,61 @@ def add_reconnect_callback(self, callback: Callable[[], None]):
 
         self._reconnect_callbacks.add(callback)
 
+    async def __cleanup_connection(self, exc):
+        await asyncio.gather(
+            self.connection.close(exc),
+            return_exceptions=True,
+        )
+        self.connection = None
+
     async def connect(self, timeout: TimeoutType = None, **kwargs):
+        if self.is_closed:
+            raise RuntimeError("{!r} connection closed".format(self))
```

Comment:
This method is for connecting, maybe raising exception here is not needed?

Reply:
Or it is should be at `reconnect` method.

Reply:
Nope. It's a case when the user wants to reopen closed connection.

Reply:
Do you mean that if robust connection was closed then the user can't open it again?

Reply:
`self.is_closed` will be changed only after `self.close()` called. But never in reconnecting state.

Reply:
I understand it. I mean following:
```python
connection = await aio_pika.connect_robust(...)
await connection.close()
await connection.connect()  # This would be illegal?
<RuntimeError>
```

Reply:
Yep illegal